keps/sig-multicluster/2149-clusterid/README.md

```diff
@@ -0,0 +1,705 @@
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+The new multi-cluster services API (see [KEP-1645](https://github.com/kubernetes/enhancements/tree/master/keps/sig-multicluster/1645-multi-cluster-services-api)) 
+expanded the ways clusters can communicate with each other and organized them into `ClusterSet`s, but as of now there is no way for a cluster to be uniquely identified 
+in a Kubernetes-native way. This document by SIG-Multicluster proposes a standard 
+for how cluster IDs should be stored and managed, based on concrete use cases 
+discussed and observed in `ClusterSet` deployments. 
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+That there must be some way to identify individual clusters in a multi-cluster 
+deployment has felt like a given to SIG-Multicluster; it has been discussed in
+a broad sense previously ([see this doc](https://docs.google.com/document/d/1F__vEKeI41P7PPUCMM9PVPYY34pyrvQI5rbTJVnS5c4/edit?usp=sharing), and was scoped 
+down in response to actual observed use cases in the latest community discussion on
+which this KEP is based ([doc](https://docs.google.com/document/d/1S0u6xzP2gcJKPipA6tBNDNuid76nVKeGhTk7PrCIuQY/edit?usp=sharing)). The motivation
+of this KEP is to provide a flexible but useful baseline for clusterID that can
+work with the known use cases.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+* Propose a standard for how cluster identification metadata should be stored and 
+managed as Kubernetes resources
```

Comment:
👋 Hi there, I stumbled upon this KEP and was wondering how we can collaborate from the [Cluster API](https://cluster-api.sigs.k8s.io) project (under SIG Cluster Lifecycle).

There are some parts of this proposal that we might collaborate on, especially around how we can identify groups of clusters. There is other parts, like the identification bits, which I might need more time to reason about, given that we have ways today to identify clusters in Cluster API through the `Cluster` resource. 

Most of the team and community members are going on vacation until the new year, I'm happy to come to one of SIG-MultiCluster meetings to engage a bit more, would that be ok with you folks?

Reply:
Please do - SIG-MC meetings are a great place for discussion